prediction-applier/src/integration-test/java/org/alfresco/hxi_connector/prediction_applier/adapters/auth/PredictionApplierHxAuthClientIntegrationTest.java

```diff
@@ -25,23 +25,78 @@
  */
 package org.alfresco.hxi_connector.prediction_applier.adapters.auth;
 
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.aResponse;
+import static com.github.tomakehurst.wiremock.client.WireMock.givenThat;
+import static com.github.tomakehurst.wiremock.client.WireMock.post;
+import static com.github.tomakehurst.wiremock.client.WireMock.postRequestedFor;
+import static com.github.tomakehurst.wiremock.client.WireMock.urlPathEqualTo;
+import static org.apache.hc.core5.http.ContentType.APPLICATION_FORM_URLENCODED;
+import static org.apache.hc.core5.http.HttpHeaders.HOST;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.then;
+
+import com.github.tomakehurst.wiremock.client.WireMock;
+import com.github.tomakehurst.wiremock.matching.EqualToPattern;
+import org.apache.camel.Exchange;
+import org.apache.hc.core5.http.HttpStatus;
+import org.junit.jupiter.api.Test;
 import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
 import org.springframework.boot.test.context.SpringBootTest;
 import org.springframework.retry.annotation.EnableRetry;
+import org.springframework.test.context.DynamicPropertyRegistry;
+import org.springframework.test.context.DynamicPropertySource;
+import org.testcontainers.junit.jupiter.Container;
 import org.testcontainers.junit.jupiter.Testcontainers;
+import org.wiremock.integrations.testcontainers.WireMockContainer;
 
+import org.alfresco.hxi_connector.common.adapters.auth.AuthenticationResult;
 import org.alfresco.hxi_connector.common.adapters.auth.HxAuthenticationClientTest;
-import org.alfresco.hxi_connector.prediction_applier.config.HxInsightProperties;
-import org.alfresco.hxi_connector.prediction_applier.config.SecurityConfig;
+import org.alfresco.hxi_connector.common.adapters.auth.util.AuthUtils;
+import org.alfresco.hxi_connector.common.test.docker.util.DockerContainers;
 
-@SpringBootTest(
-        classes = {HxInsightProperties.class, SecurityConfig.class},
-        properties = "logging.level.org.alfresco=DEBUG")
+@SpringBootTest(properties = "logging.level.org.alfresco=DEBUG")
 @EnableAutoConfiguration
 @EnableConfigurationProperties
 @EnableRetry
 @Testcontainers
 @SuppressWarnings("PMD.TestClassWithoutTestCases")
 class PredictionApplierHxAuthClientIntegrationTest extends HxAuthenticationClientTest
-{}
+{
+    @Container
+    private static final WireMockContainer ACS_MOCK = DockerContainers.createWireMockContainer();
+
+    @Test
+    protected void testAuthorize_alfresco()
```

Comment:
This method will likely be placed in the parent test class as it tests the newly added `authenticate` method.